File: Frontend/src/services/profileService.js
import axiosInstance from "../api/axiosInstance";

export const profileService = {

    getProfile: async () => {

        const response = await axiosInstance.get("/customer/profile");

        return response.data;
    },

    updateProfile: async (profileData) => {

        const response = await axiosInstance.put(
            "/customer/profile",
            profileData
        );

        return response.data;
    }

};

File: Backend/src/main/java/com/project/autoserve/service/ProfileService.java
package com.project.autoserve.service;

import com.project.autoserve.dto.profile.CustomerProfileResponseDTO;
import com.project.autoserve.dto.profile.UpdateProfileRequestDTO;

public interface ProfileService {

    CustomerProfileResponseDTO getProfile();

    CustomerProfileResponseDTO updateProfile(
            UpdateProfileRequestDTO request);

}